docs/utils/navigation.utils.ts

```diff
@@ -147,11 +147,20 @@ export function handleHrefClickEventWithRouter(e: Event, router: Router) {
     return;
   }
 
-  const relativeUrl = closestAnchor?.getAttribute?.('href');
-  if (relativeUrl) {
-    e.preventDefault();
-    router.navigateByUrl(relativeUrl);
+  const hrefAttr = closestAnchor?.getAttribute?.('href');
+  let relativeUrl: string;
+  if (hrefAttr?.startsWith('http')) {
+    // Url isn't relative but we're targeting the same domain
```

Comment:
How are you determining that it’s the same domain?

Reply:
Found it, it’s called responsibly to filter that. 

NIT: but would be clearer if the function is self contained and not rely on the caller to perform this filtering. As from re-usability perspective, it makes this function very error prone.

Reply:
agreed.